### PR TITLE
[4794][IMP] account_commission_sales_fee

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[4794](https://www.quartile.co/web#id=4794&cids=3&menu_id=505&action=1457&model=project.task&view_type=form)

### CODE REVIEW

File: account_commission_sales_fee/models/commission_settlement.py [hunk 1]
```diff
@@ -11,7 +11,10 @@ def _prepare_sales_fee_product(self, commission):
         return {
             "product_id": commission.sales_fee_product_id.id,
             "quantity": len(
-                self.line_ids.filtered(lambda line: line.commission_id == commission)
+                self.line_ids.filtered(
+                    lambda line: line.commission_id == commission
+                    and line.invoice_line_id.move_id.move_type == "out_invoice"
+                )
```
Comment (yostashiro) [inline account_commission_sales_fee/models/commission_settlement.py]: I don't think we should rely on move_type as out_invoice could also hold lines with negative subtotals.
```suggestion
                self.line_ids.filtered(
                    lambda line: line.commission_id == commission and line.settled_amount > 0.0
                )
```

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: account_commission_sales_fee/models/commission_settlement.py (hunk 1 above)
Comment (AungKoKoLin1997) [reply account_commission_sales_fee/models/commission_settlement.py]: I updated the code.

**Review by yostashiro — APPROVED** [review]
Code review.

**Review by yostashiro — CHANGES_REQUESTED** [review]
Sorry, I approved too early.

@AungKoKoLin1997 Can you please sort out the test error?

**Review by yostashiro — COMMENTED** [review]
@AungKoKoLin1997 Is the test fix related to the update of the code this time? I couldn't tell.

**Comment by AungKoKoLin1997** [comment]
> @AungKoKoLin1997 Is the test fix related to the update of the code this time? I couldn't tell.

Yes. It is. Now we added `settled_amount` in filtered. With pervious code, the settlement lines is 2 lines but `settled_amount` are `0.0` because the `amount` in the agent line of invoice line is 0.0. It is because of the base amount is not within the range of section_ids. https://github.com/qrtl/rmm-oca/blob/b88195713c3f92f3a30364de5d374c59ef94eae0/commission/models/commission.py#L40-L45.
So, instead of checking how section_ids was defined and use the amount within this range, I decided to change commission type.

**Review by yostashiro — APPROVED** [review]

**Comment by yostashiro** [comment]
@AungKoKoLin1997 Thanks Lin. I think the tests could be improved but it's OK for now.

**Review by kanda999 — APPROVED** [review]
Functional Check

**Review by yostashiro — APPROVED** [review]

**Pull request merged**